pkg/sql/pg_extension.go

```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
 	validWithNoDatabaseContext: false,
 }
 
+func postgisColumnsTablePopulator(
+	matchingFamily types.Family,
+) func(context.Context, *planner, *DatabaseDescriptor, func(...tree.Datum) error) error {
+	return func(ctx context.Context, p *planner, dbContext *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		return forEachTableDesc(
+			ctx,
+			p,
+			dbContext,
+			hideVirtual,
+			func(db *sqlbase.DatabaseDescriptor, scName string, table *sqlbase.TableDescriptor) error {
+				if !table.IsPhysicalTable() {
+					return nil
+				}
+				for _, colDesc := range table.Columns {
+					if colDesc.Type.Family() != matchingFamily {
+						continue
+					}
+					m := colDesc.Type.InternalType.GeoMetadata
```

Comment:
I think generally we don't expose access to InternalType

Reply:
fixed.